### ISSUE TITLE
Fill Tesssellator Changes Custom Attributes on Intersections

### ISSUE THREAD

**Issue by anlumo** [nical/lyon#699]
I'm implementing my own stroke tessellator (because the one in lyon isn't flexible enough for my needs). For this, I'm defining the outline of the stroke with the advancement and line side as custom attributes and then let the fill tessellator have a go at it so I get triangles.

However, when the path self-intersects (which is a completely valid situation for my application), it looks like this: 
![image](https://user-images.githubusercontent.com/644763/132774177-9ef7e2e4-1d29-4034-966d-81dbde549db9.png)

Green is the left side of the stroke, red is the right side (sorry to color-blind people). The problem is that where the left side of one segment intersects with the right side of the other segment, I get yellow, which is completely not what I want. Is there a way to stop the fill tessellator to not interpolate between the attributes of vertices of unconnected segments? I think the solution would need the fill tessellator to not merge these two parts at all. I saw that there's an option to not handle self-intersections, but the documentation warns to never turn this off if there are self-intersections.

**Comment by nical** [comment]
The tessellator emits a single point at self-intersections so it can't give you exactly what you want here. Changing this behavior to produce multiple vertices at the intersection is probably possible but it would be a pretty major rewrite of a good chunk of the algorithm, I don't have the bandwidth to give it a go any time soon.

When turning off handling self-intersections, the algorithm will not add any vertex at the intersection and will do a best effort not to panic during the tessellation but it will produce incorrect results when there are self-intersections so it's likely not what you want either.

**Comment by anlumo** [comment]
The basic issue is that it's not really an intersection if the attributes differ too much, even though they're spatially in the same place. I'm aware that this might be impossible to solve for a generic fill tessellator, but I figured that I'd mention it anyways.

**Comment by nical** [comment]
Yeah I totally see where you are coming from and I appreciate better understanding how people use lyon and what could make it more useful. This is a particularly tricky one, though, as you figured.

**Comment by nical** [comment]
Handling in the fill tessellator would require a major rewrite so I'm going to close this as out of scope, sorry. We can tryto get the stroke tessellator closer to handling your use case, though.
